File: algorithm/js/8.remove-duplicage-element-from-array-without-new-array.js
/**
 * Remove duplicate element from arry
 */

const arr = [1, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6]

function remove_duplicate(arr) {

  return [...new Set(arr)].length
}

console.log(remove_duplicate(arr))

File: algorithm/python/8.remove-duplicage-element-from-array-without-new-array.py
"""
Remove duplicate element from array without new array
"""

arr = [1, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6]

def remove_duplicate(arr):
  length = len(arr)
  newArr = []
  for i in range(length):
    if arr[i-1] != arr[i]:
      newArr.append(arr[i])
  
  print(newArr)
  return len(newArr)


print(remove_duplicate(arr))